tiledb/CMakeLists.txt

```diff
@@ -593,51 +593,52 @@ if(TILEDB_SERIALIZATION)
-    list(APPEND CAPNP_COMPILE_COMMAND ${CAPNP_EXECUTABLE} compile -oc++:"${CMAKE_CURRENT_BINARY_DIR}" ${CMAKE_CURRENT_SOURCE_DIR}/sm/serialization/tiledb-rest.capnp --src-prefix ${CMAKE_CURRENT_SOURCE_DIR})
+    set(TILEDB_SERIALIZATION_GENERATED_SUBDIR posix)
   endif()
 
-  # note: run the compiler under `cmake -E env ...` because we need to run capnp executable
-  #       the capnp driver needs to be able to find the plugin executables (eg capnpc-c++)
-  message(STATUS "CAPNP_COMPILE_COMMAND=${CAPNP_COMPILE_COMMAND}")
-  add_custom_command(
-	    DEPENDS ${TILEDB_CORE_INCLUDE_DIR}/tiledb/sm/serialization/tiledb-rest.capnp
-      OUTPUT
-        ${CMAKE_CURRENT_BINARY_DIR}/sm/serialization/tiledb-rest.capnp.c++
-        ${CMAKE_CURRENT_BINARY_DIR}/sm/serialization/tiledb-rest.capnp.h
-      WORKING_DIRECTORY
-        ${TILEDB_EP_BASE}/..
-      COMMAND
-        "${CAPNP_COMPILE_COMMAND}"
-      COMMAND_EXPAND_LISTS
-      COMMENT "(re-)generate tiledb-rest.capnp.<h,c++>"
-    )
-
-  set_source_files_properties(
-    "${CMAKE_CURRENT_BINARY_DIR}/sm/serialization/tiledb-rest.capnp.c++"
-    "${CMAKE_CURRENT_BINARY_DIR}/sm/serialization/tiledb-rest.capnp.h"
-    PROPERTIES GENERATED TRUE
-    )
-
-  set_property(SOURCE ${CMAKE_CURRENT_BINARY_DIR}/sm/serialization/tiledb-rest.capnp.c++
-	  APPEND PROPERTY OBJECT_DEPENDS ${TILEDB_CORE_INCLUDE_DIR}/tiledb/sm/serialization/tiledb-rest.capnp
-	)
-  set_property(
-    SOURCE ${CMAKE_CURRENT_BINARY_DIR}/sm/serialization/tiledb-rest.capnp.h
-	  APPEND PROPERTY OBJECT_DEPENDS ${TILEDB_CORE_INCLUDE_DIR}/tiledb/sm/serialization/tiledb-rest.capnp
-	)
-
   target_sources(TILEDB_CORE_OBJECTS PUBLIC
-    ${CMAKE_CURRENT_BINARY_DIR}/sm/serialization/tiledb-rest.capnp.c++
+    ${CMAKE_CURRENT_SOURCE_DIR}/sm/serialization/${TILEDB_SERIALIZATION_GENERATED_SUBDIR}/tiledb-rest.capnp.c++
   )
   if (WIN32 AND TILEDB_STATIC)
     target_sources(TILEDB_CORE_OBJECTS_STATIC PUBLIC
-      ${CMAKE_CURRENT_BINARY_DIR}/sm/serialization/tiledb-rest.capnp.c++
+      ${CMAKE_CURRENT_SOURCE_DIR}/sm/serialization/${TILEDB_SERIALIZATION_GENERATED_SUBDIR}/tiledb-rest.capnp.c++
     )
   endif()
+
+  ###################################
+  # capnproto file generation target
+  ###################################
+
+  # This target regenerates the checked-in generated source files for serialization
+  # support. It must be re-run whenever `tiledb-rest.capnp` is updated.
+
+  set(TILEDB_SERIALIZATION_GENERATED_DIR
+      ${CMAKE_CURRENT_SOURCE_DIR}/sm/serialization/${TILEDB_SERIALIZATION_GENERATED_SUBDIR})
+
+  # We only need to override the include path and PATH env for EP builds
+  if(TILEDB_CAPNP_EP_BUILT)
+    list(APPEND CAPNP_COMPILE_COMMAND ${CMAKE_COMMAND} -E env PATH=${TILEDB_EP_BASE}/install/bin ${CAPNP_EXECUTABLE} compile -I "${TILEDB_EP_BASE}/install/include" -oc++:"${TILEDB_SERIALIZATION_GENERATED_DIR}" ${CMAKE_CURRENT_SOURCE_DIR}/sm/serialization/tiledb-rest.capnp --src-prefix=${CMAKE_CURRENT_SOURCE_DIR}/sm/serialization)
```

Comment:
Do we need quotes here for the `${CMAKE_CURRENT_SOURCE_DIR}/sm/serialization/tiledb-rest.capnp` and `--src-prefix=${CMAKE_CURRENT_SOURCE_DIR}/sm/serialization` ?

Reply:
Updated all, safer that way. I only tested in a non-whitespace path on windows.